### PR TITLE
Updated error message to include server side errors if present.

### PR DESCRIPTION
**Pull request by neshdev**
Error message to include server side error messages if present in payload.

BUG=b/375194356

### CODE REVIEW

File: src/kagglehub/exceptions.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def kaggle_api_raise_for_status(response: requests.Response, resource_handle: Op
         response.raise_for_status()
     except requests.HTTPError as e:
         message = str(e)
+        server_error_message = response.json().get("message", "")
```
Comment (dolaameng) [inline src/kagglehub/exceptions.py:61]: I guess some response are not json format so it fails some integration tests?

Comment (mohami2000) [reply src/kagglehub/exceptions.py:61]: +1

File: src/kagglehub/exceptions.py [hunk 2]
```diff
@@ -58,9 +58,11 @@ def kaggle_api_raise_for_status(response: requests.Response, resource_handle: Op
         response.raise_for_status()
     except requests.HTTPError as e:
         message = str(e)
-        server_error_message = response.json().get("message", "")
-        if server_error_message:
-            server_error_message = f"The server reported the following issues: {server_error_message}\n"
+        server_error_message = ""
+        if response.headers.get("Content-Type") == "application/json":
```
Comment (rosbo) [inline src/kagglehub/exceptions.py:62]: Should probably do a case insensitive check. 

add a try/except block to catch the `requests.exception.JSONDecodeError`: https://requests.readthedocs.io/en/latest/user/quickstart/#json-response-content

This way, if the mime type is `application/json` but the response is invalid, we won't break here.

**Review by rosbo — CHANGES_REQUESTED** [review]

File: src/kagglehub/exceptions.py (hunk 2 above)
Comment (neshdev) [reply src/kagglehub/exceptions.py]: Skipping the header check all together and just using a try/catch instead.

**Review by rosbo — APPROVED** [review]

**Review by mohami2000 — APPROVED** [review]

**Pull request merged**